src/components/Form/Input/CurrencyInput.tsx

```diff
@@ -1,31 +1,199 @@
-import { useState, useCallback, useEffect } from 'react'
-import { getValueWithDecimalFormat, unFormatCurrency } from 'dd360-utils'
-import BaseInput, { InputProps } from './BaseInput'
-
-function CurrencyInput(props: InputProps) {
-  const { onChange, value, prefix = '' } = props
-  const [localValue, setLocalValue] = useState(
-    getValueWithDecimalFormat(String(value))
-  )
-
-  const handleChange = useCallback(
-    (event: React.ChangeEvent<HTMLInputElement>) => {
-      event.target.value = getValueWithDecimalFormat(event.target.value)
-      setLocalValue(event.target.value)
-      event.target.value = unFormatCurrency(event.target.value).toString()
-      onChange && onChange(event)
-    },
-    [onChange, localValue]
-  )
-
-  useEffect(() => {
-    if (!isNaN(Number(value)))
-      setLocalValue(getValueWithDecimalFormat(String(value)))
-  }, [value])
-
-  return (
-    <BaseInput {...props} onChange={handleChange} value={prefix + localValue} />
-  )
+import { useState, useCallback, useRef, forwardRef } from 'react'
+import CInput from 'react-currency-input-field'
+import { composeClasses } from 'lib/classes'
+import { getPaddingInput } from '../shared'
+import FormLabel from '../FormLabel'
+import { IconStatus, InputProps } from './BaseInput'
+import useInputStyles from './useInputStyles'
+import { unFormatCurrency } from 'dd360-utils'
+
+export interface InputCurrencyProps extends InputProps {
+  language?: 'es' | 'en'
+  prefix?: string
+  groupSeparator?: string
+  decimalSeparator?: string
+  decimalsLimit?: number
+  suffix?: string
+}
+
+const getEvent = (e: React.FocusEvent<HTMLInputElement>, prefix: string) => {
+  const newValue = e.target.value.replace(prefix, '')
+  const unformatedValue: any = unFormatCurrency(newValue)
+  return {
+    ...e,
+    target: {
+      ...e.target,
+      value: unformatedValue || undefined,
+      name: e.target.name
+    }
+  }
 }
 
+const CurrencyInput = forwardRef<HTMLDivElement, InputCurrencyProps>(
+  (
+    {
+      variant = 'default',
+      label,
+      rounded = 'lg',
+      className,
+      classNameAdornment,
+      padding,
+      paddingX = '4',
+      paddingY,
+      startAdornment,
+      endAdornment,
+      message,
+      inputBlank,
+      onFocus,
+      onBlur,
+      large,
+      boxShadow = 'lg',
+      style,
+      placeholder,
+      value,
+      isRequired,
+      prefix = '$',
+      suffix,
+      groupSeparator = ',',
+      decimalSeparator = '.',
+      decimalsLimit = 2,
+      min,
+      max,
+      onChange,
+      disabled,
+      internalClassName,
```

Comment:
why we need an internal className? 🤔

Reply:
we use it to style the input and not the container, that is so that it fits well when we use it in the tables